Source/automap_render.cpp

```diff
@@ -0,0 +1,84 @@
+#include "automap_render.hpp"
+
+namespace devilution {
+namespace {
+
+enum class DirectionX {
+	EAST = 1,
+	WEST = -1,
+};
+
+enum class DirectionY {
+	SOUTH = 1,
+	NORTH = -1,
+};
+
+template <DirectionX DirX, DirectionY DirY>
+void DrawMapLine(const CelOutputBuffer &out, Point from, int height, std::uint8_t colorIndex)
+{
+	while (height-- > 0) {
+		SetPixel(out, from, colorIndex);
+		from.x += static_cast<int>(DirX);
+		SetPixel(out, from, colorIndex);
+		from.x += static_cast<int>(DirX);
+		from.y += static_cast<int>(DirY);
+	}
+	SetPixel(out, from, colorIndex);
+}
+
+template <DirectionX DirX, DirectionY DirY>
+void DrawMapLine2(const CelOutputBuffer &out, Point from, int width, std::uint8_t colorIndex)

```

Comment:
its a bit hard to understand what the difference between the regular and `2` functions are. from looking here im guessing its because you need to alternate at have lines line up when drawing them in blocks?

Reply:
The '2' function increments y twice per x increment, and the 'plain' function increments x twice per y increment. 

Maybe call them "Steep" and "Flat" or "Shallow"?

Reply:
Done, the "2" functions are now "Steep", thanks for the suggestion!